sqlglot/dialects/tsql.py

```diff
@@ -671,19 +671,20 @@ def create_sql(self, expression: exp.Create) -> str:
 
             if exists:
                 identifier = self.sql(exp.Literal.string(exp.table_name(table) if table else ""))
+                sql = exp.Literal.string(sql).sql()
```

Comment:
sql = self.sql(exp.Literal.string(...))

you need the sql generation to be dialect specific